### DISCUSSION TITLE
Adding support for Wayland

### DISCUSSION THREAD

**Discussion by dostological** [lukas-blecher/LaTeX-OCR#36] [Ideas]
Hello. I am very interested in this project and I think it's awesome, as a student, this would help me tremendously. Unfortunately, it appears that this does not support Wayland (on gnome for linux), but only X. Here is what using the gui outputs:

> Warning: Ignoring XDG_SESSION_TYPE=wayland on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway.
/home/archy/.local/lib/python3.9/site-packages/torch/cuda/__init__.py:106: UserWarning: 
NVIDIA GeForce RTX 3050 Ti Laptop GPU with CUDA capability sm_86 is not compatible with the current PyTorch installation.
The current PyTorch install supports CUDA capabilities sm_37 sm_50 sm_60 sm_70.
If you want to use the NVIDIA GeForce RTX 3050 Ti Laptop GPU GPU with PyTorch, please check the instructions at https://pytorch.org/get-started/locally/

>  warnings.warn(incompatible_device_warn.format(device_name, capability, " ".join(arch_list), device_name))
Traceback (most recent call last):
  File "/home/archy/Programs/LaTeX-OCR/gui.py", line 252, in mouseReleaseEvent
    img = ImageGrab.grab(bbox=(x1, y1, x2, y2), all_screens=True)
  File "/usr/lib/python3.9/site-packages/PIL/ImageGrab.py", line 64, in grab
    size, data = Image.core.grabscreen_x11(xdisplay)
OSError: X get_image failed: error 8 (73, 0, 678)
zsh: abort (core dumped)  python gui.py

**Comment by heyeshuang** [comment]
It looks like Pillow/ImageGrab doesn't support Wayland. The original `pix2tex.py` may work. 

See <https://pillow.readthedocs.io/en/latest/reference/ImageGrab.html> and <https://stackoverflow.com/questions/44104331/how-to-capture-screen-on-waylandgnome-in-python-code>

**Reply by lukas-blecher** [reply]
Thanks for your input @heyeshuang. You are right, the non-gui version `pix2tex.py` can also work with images that are saved on the hard disk.
@dostological you would have to find a working snipping tool for your OS and enter the path to the image into the terminal.

**Comment by lukas-blecher** [comment]
This is now supported. You need to install `gnome-screenshot` and pass the `--gnome` flag to the gui:
```
python gui.py --gnome
```
